wordpress-mobile/WordPress-Android
# [Blogging prompts v1] Implement prompt reminder notification analytics events

Fixes #16521 

To test:
Enable feature flag:
- Open Debug settings
- `blogging_prompts_notification_remote_field` should not be listed
- enable `blogging_prompts_remote_field`

With feature flag enabled:
- On `My Site`, tap on `Menu`
- Tap on `Site Settings`
- Tap on `Reminders and prompts` and select a date and time for the reminder
- Change the device date to something after the selected date/time for the reminder
- A notification to answer the prompt should appear
- If the feature flag mentioned above is disabled, the existing blogging reminder notification (not related to blogging prompts) should be shown 
    - `"blogging_reminders_notification_received"` should de tracked (with parameter `"prompt_included": "true"`) when notification is received
    - `"blogging_reminders_notification_prompt_answer_tapped"` should be tracked when "Answer" button is tapped
    - `"blogging_reminders_notification_prompt_dismiss_tapped"` should be tracked when "Dismiss" button is tapped
    - `"blogging_reminders_notification_prompt_tapped"` should be tracked when notification is tapped directly
    - `"blogging_reminders_notification_prompt_dismissed"` should be tracked when notification is dismissed not using "Dismiss" button (e.g. swipe)

## Regression Notes
1. Potential unintended areas of impact
Blogging reminder notification with blogging prompts feature flag enabled

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and updated unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated `BloggingRemindersAnalyticsTrackerTest.kt` and `ReminderNotifierTest.kt`

PR submission checklist:
